local-tests/tests/testSignTransactionWithSolanaEncryptedKey.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  AccessControlConditions,
+  ILitNodeClient,
+  LitAbility,
+} from '@lit-protocol/types';
+import { encryptString } from '@lit-protocol/lit-node-client-nodejs';
+import {
+  Connection,
+  Keypair,
+  LAMPORTS_PER_SOL,
+  PublicKey,
+  SystemProgram,
+  Transaction,
+  clusterApiUrl,
+} from '@solana/web3.js';
+import {
+  LitAccessControlConditionResource,
+  LitActionResource,
+  createSiweMessageWithRecaps,
+  generateAuthSig,
+} from '@lit-protocol/auth-helpers';
+
+/**
+ * Test Commands:
+ * ✅ NETWORK=cayenne yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ * ✅ NETWORK=manzano yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ * ❔ NETWORK=localchain yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ */
+export const testSignTransactionWithSolanaEncryptedKey = async (
+  devEnv: TinnyEnvironment
+) => {
+  const alice = await devEnv.createRandomPerson();
+  const solanaKeypair = Keypair.generate();
+  const decryptionAccessControlConditions: AccessControlConditions = [
+    {
+      contractAddress: '',
+      standardContractType: '',
+      chain: 'ethereum',
+      method: '',
+      parameters: [':userAddress', 'latest'],
+      returnValueTest: {
+        comparator: '=',
+        value: await alice.wallet.getAddress(),
+      },
+    },
+  ];
+
+  const { ciphertext, dataToEncryptHash } = await encryptString(
+    {
+      accessControlConditions: decryptionAccessControlConditions,
+      dataToEncrypt: Buffer.from(solanaKeypair.secretKey).toString('hex'),
+    },
+    devEnv.litNodeClient as unknown as ILitNodeClient
+  );
+
+  const sessionSigs = await devEnv.litNodeClient.getSessionSigs({
+    chain: 'ethereum',
+    resourceAbilityRequests: [
+      {
+        resource: new LitActionResource('*'),
+        ability: LitAbility.LitActionExecution,
+      },
+      {
+        resource: new LitAccessControlConditionResource('*'),
+        ability: LitAbility.AccessControlConditionDecryption,
+      },
+    ],
+    authNeededCallback: async ({ uri, resourceAbilityRequests }) => {
+      const toSign = await createSiweMessageWithRecaps({
+        uri: uri!,
+        expiration: new Date(Date.now() + 1000 * 60 * 10).toISOString(),
+        resources: resourceAbilityRequests!,
+        walletAddress: await alice.wallet.getAddress(),
+        nonce: await devEnv.litNodeClient!.getLatestBlockhash(),
+        litNodeClient: devEnv.litNodeClient,
+      });
+      return await generateAuthSig({
+        signer: alice.wallet,
+        toSign,
+      });
+    },
+  });
+
+  const solanaTransaction = new Transaction();
+  solanaTransaction.add(
+    SystemProgram.transfer({
+      fromPubkey: solanaKeypair.publicKey,
+      toPubkey: new PublicKey(solanaKeypair.publicKey),
+      lamports: LAMPORTS_PER_SOL / 100, // Transfer 0.01 SOL
+    })
+  );
+  solanaTransaction.feePayer = solanaKeypair.publicKey;
```

Comment:
What all should be part of the existing interface `SolanaLitTransaction`?

Reply:
Solana txs appear to be very different from ETH txs. It seems there are specific instructions depending on what type of transaction you're trying to submit, and they have different requirements. Because of this, I don't think we should try to create a `SolanaLitTransaction`, and what I've opted for is for the user to just pass their serialized transaction that will be signed within the Lit Action (that way the Lit Action doesn't need to care about the type of tx being sent, it just signs it arbitrarily)

<img width="824" alt="image" src="https://github.com/LIT-Protocol/js-sdk/assets/20375610/b8208539-64b2-480a-8dd3-8c4d6d9863f3">

Reply:
Sure Eth tx have other fields too: https://docs.ethers.org/v5/api/providers/types/#providers-TransactionRequest
But only some fields are mandatory in Solana just like Eth and the rest are autofilled like in Ethers?
What I meant was adding only the mandatory ones, which you're using above: `fromPubkey`, `toPubkey`, `lamports` and `feePayer` to `SolanaLitTransaction`
What do you think about it?

Reply:
> autofilled like in Ethers

The Solana SDK will autofill some of the properties, but you still have to instantiate one of the above `SystemProgram`s and these all require specific params. Those parameters I'm using in my tests are what's required to send a `transfer` `SystemProgram`, but  just using those same properties won't mean you can send the other types of txs

Reply:
`SolanaLitTransaction` should accept the serialized Solana tx provided by the user